cluster-autoscaler/core/scale_down.go

```diff
@@ -518,7 +518,7 @@ func (sd *ScaleDown) UpdateUnneededNodes(
 		additionalNodesToRemove, additionalUnremovable, additionalNewHints, simulatorErr :=
 			simulator.FindNodesToRemove(currentNonCandidates[:additionalCandidatesPoolSize], destinationNodes, nonExpendablePods, nil,
 				sd.context.PredicateChecker, additionalCandidatesCount, true,
-				sd.podLocationHints, sd.usageTracker, timestamp, pdbs)
+				sd.podLocationHints, sd.usageTracker, timestamp, pdbs, emptyNodesList)
```

Comment:
Why pass `emptyNodesList` and not `emptyNodes`? Map would be more convenient to use (no need to loop over the list searching for each node)

Reply:
The reason was that `simulator.FindNodesToRemove` is also invoked in `TryToScaleDown` where there is only `emptyNodes`, which is a `[]*apiv1.Node`, not a map. Should I maybe introduce another helper function to generate this map out of the list and also invoke it in `TryToScaleDown` to get the map?

Reply:
Hey, sorry, missed the emails from github :/ I'd say yes, let's put it into map. It's probably irrelevant performance-wise for all but the largest clusters, but it seems cleaner.

Reply:
Changed & rebased the branch.